docs/classes.md

```diff
@@ -0,0 +1,263 @@
+# Supported Class by Pyccel
+
+Pyccel strives to provide robust support for object-oriented programming concepts commonly used by developers. In Pyccel, classes are a fundamental building block for creating structured and reusable code. This documentation outlines key features and considerations when working with classes in Pyccel.
+
+## Constructor Method
+
+-   The Constructor Method, `__init__`, is used to initialize the object's attributes.
+-   It must be named `__init__`, or the Pyccel compiler will generate an error.
```

Comment:
What error are you talking about here? Normally it is not possible to initialise an object's attributes with a function with a different name. Is your point that you have to have an `__init__` method or that you cannot assign to attributes in other methods if they have not first been created in the `__init__` method?

Reply:
Fixed https://github.com/pyccel/pyccel/pull/1580/commits/f92c3e1883b231711e3cbad1f475439610d22f96.